src/java.base/share/classes/jdk/internal/reflect/Reflection.java

```diff
@@ -53,6 +53,9 @@ public class Reflection {
     public static final Set<String> ALL_MEMBERS = Set.of(WILDCARD);
 
     static {
+        // 3 filter scenarios:
+        // 1. Classes loaded before Reflection, may (System) or may not
+        //    (ConstantPool) be initialized before main call: below
```

Comment:
This comment is confusing. I assume you want to say that these classes may or may not be loaded before the application main method. It may be simpler to just drop the proposed comments as they beg too many questions.

Reply:
Let's put these 4 points in time: class A load, A init, Reflection init, main run. (If an event is after "main run", it may be completely optional)

There can be these arrangements:
1. A load, Reflection init, main run; subcases include:
   a. A load, A init, Reflection init, main run (System)
   b. A load, Reflection init, A init, main run (none known yet)
   c. A load, Reflection init, main run, A init (ConstantPool)
   Handling: register in Reflection static block
2. Reflection init, A load, A init, main run (MethodHandles.Lookup)
   Handling: register in their own static block
3. Reflection init, main run, A init; subcases incldue:
   a. Reflection init, A load, main run, A init (none known yet)
   b. Reflection init, main run, A load, A init (sun.misc.Unsafe)
   Handling: Quite hard, easily causing extra class loading

How should I phrase my comments?